src/lib/operations/encrypt/encrypt.handler.ts

```diff
@@ -0,0 +1,42 @@
+import express from "express";
+import { encrypt } from "./encrypt";
+import { ExpressRequestWithToken } from "../../utils/auth/auth0";
+import { Route } from "../../server/server.constants";
+import { logRequestSuccess, logRequestStart } from "../../utils/log/log.utils";
+
+export interface EncryptIdTokenData {
+  keyName: string;
+  plaintext: string;
+}
+
+export function encryptHandlerFactory() {
+  return async (
+    req: ExpressRequestWithToken<EncryptIdTokenData>,
+    res: express.Response,
+  ) => {
+    try {
+      const { idToken } = req;
+      const { data } = idToken;
+
+      // TODO: Replace with Joi.
+      if (!idToken || !data || !data.keyName || !data.plaintext) {
+        throw new Error("Invalid JWT");
+      }
+
+      logRequestStart(Route.ENCRYPT, idToken);
+
+      const ciphertext = await encrypt(data.plaintext, data.keyName);
+
+      // TODO: Do not log this!
+      logRequestSuccess(Route.ENCRYPT);
```

Comment:
Before, we were logging the responses, which could include ciphertext, plaintext, signatures... I don't think that should be the case, the same way we are not logging the ID token as-is. Now, we just log the same ID token to be able to group log messages.